### PR TITLE
Task ADCM-1133

### PR DESCRIPTION
**Pull request by Maunty**
Packer changes:
Parameters of function build became all named bcs one of unnamed parameter became unnecessary.
In order to save backward compatibility i made all parameters named and saved their order in function declaration
Console option -n became required=False.
Added input parameters analization of build function call with default values asigment.
Function adcm_client.packer.bundle_build.build became more suitable to be called from sdk - it returns dict with tarballs full path and bytes stream as values.
Tarball saving on fs moved to shell program file.
Sdk changes:
adcm_client.util.stream.file refactored to use packer build function.
adcm_client.ADCM.upload_from_fs refactored in order to return BundleList object of loaded bundles using new stream.file.

### CODE REVIEW

File: adcm_client/objects.py [hunk 1]
```diff
@@ -937,8 +937,16 @@ def _upload(self, bundle_stream) -> Bundle:
         return self.bundle(bundle_id=data['id'])
 
     @allure_step('Upload bundle from "{1}"')
-    def upload_from_fs(self, dirname) -> Bundle:
-        return self._upload(stream.file(dirname))
+    def upload_from_fs(self, dirname, **args) -> BundleList:
+        streams = stream.file(dirname, **args)
```
Comment (acmnu) [inline adcm_client/objects.py]: You could change signature in that way:

```
def upload_from_fs(self, **dirnames) -> BundleList:
```

File: adcm_client/objects.py [hunk 2]
```diff
@@ -937,8 +937,16 @@ def _upload(self, bundle_stream) -> Bundle:
...
+        result = None
+        if isinstance(streams, list):
+            for st in streams:
+                result = BundleList(self._api, id=self._upload(st).id) if result is None \
+                    else result.append(BundleList(self._api, id=self._upload(st).id))
+        else:
+            result = BundleList(self._api, id=self._upload(streams).id)
```
[the remark this answers is not in the document]
Comment (acmnu) [reply adcm_client/objects.py]: Well now I am sure you could do better.

File: adcm_client/objects.py [hunk 3]
```diff
@@ -937,8 +937,16 @@ def _upload(self, bundle_stream) -> Bundle:
...
 
     @allure_step('Upload bundle from "{1}"')
-    def upload_from_fs(self, dirname) -> Bundle:
-        return self._upload(stream.file(dirname))
+    def upload_from_fs(self, dirname, **args) -> BundleList:
+        streams = stream.file(dirname, **args)
+        result = None
+        if isinstance(streams, list):
+            for st in streams:
+                result = BundleList(self._api, id=self._upload(st).id) if result is None \
```
Comment (acmnu) [inline adcm_client/objects.py]: There is no needs in that code, if you initialize result as empty BundleList, and not None.

File: adcm_client/objects.py [hunk 4]
```diff
@@ -937,8 +937,16 @@ def _upload(self, bundle_stream) -> Bundle:
...
+                    else result.append(BundleList(self._api, id=self._upload(st).id))
```
[the remark this answers is not in the document]
Comment (Maunty) [reply adcm_client/objects.py]: Fixed

File: adcm_client/objects.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Maunty) [reply adcm_client/objects.py]: Refactored

File: adcm_client/objects.py (hunk 1 above)
Comment (Maunty) [reply adcm_client/objects.py]: dirname is required parameter. **args - named parameter of func build from packer to changes its behavior.

File: adcm_client/objects.py (hunk 3 above)
Comment (Maunty) [reply adcm_client/objects.py]: Fixed

File: adcm_client/objects.py [hunk 5]
```diff
@@ -937,8 +937,19 @@ def _upload(self, bundle_stream) -> Bundle:
         return self.bundle(bundle_id=data['id'])
 
     @allure_step('Upload bundle from "{1}"')
-    def upload_from_fs(self, dirname) -> Bundle:
-        return self._upload(stream.file(dirname))
+    def upload_from_fs(self, dirname, **args) -> Bundle:
+        streams = stream.file(dirname, **args)
+        if len(streams) > 1:
+            raise TooManyArguments('Build more then one bundle from dir')
```
Comment (acmnu) [inline adcm_client/objects.py]: I don't like that message. It's better to say what to do.

> upload_bundle_from_fs is not capable with building multiple bundle editions from one dir. Use upload_from_fs_all instead.

File: adcm_client/objects.py [hunk 6]
```diff
@@ -937,8 +937,19 @@ def _upload(self, bundle_stream) -> Bundle:
...
+        return self._upload(streams[0])
+
+    @allure_step('Upload bundles from "{1}"')
+    def upload_from_fs_all(self, dirname, **args) -> BundleList:
+        streams = stream.file(dirname, **args)
+        result = BundleList(self._api, empty_bundlelist='not_existing_bundle')
```
Comment (acmnu) [inline adcm_client/objects.py]: Add comment about that dirty hack with epty_bundle_list

File: adcm_client/objects.py (hunk 5 above)
Comment (Maunty) [reply adcm_client/objects.py]: Fixed.

File: adcm_client/objects.py (hunk 6 above)
Comment (Maunty) [reply adcm_client/objects.py]: Added.

**Pull request merged**